### PR TITLE
chore: add intg and unit tests for SetResourceQuota

### PR DESCRIPTION
**Pull request by salonig23**
…9643)

## Ticket

DET-10422
DET-10419

## Description

Added intg and unit tests for SetResourceQuota
Currently blocked by https://hpe-aiatscale.atlassian.net/browse/DET-10410, which is why some licensing checking code is commented out in api_workspace.go to make sure tests pass. Before merging I will remove the commented out code and call a function from DET-10410 that will allow EE testing

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/rm/kubernetesrm/jobs.go [hunk 1]
```diff
@@ -2323,7 +2323,7 @@ func (j *jobsService) setResourceQuota(quota int, namespace string) error {
 }
 
 func (j *jobsService) getNamespaceResourceQuota(namespaceName string) (*float64, error) {
-	k8sQuotas, err := j.clientSet.CoreV1().ResourceQuotas(namespaceName).List(context.Background(), metaV1.ListOptions{})
+	k8sQuotas, err := j.clientSet.CoreV1().ResourceQuotas(namespaceName).List(context.TODO(), metaV1.ListOptions{})
```
Comment (carolinaecalderon) [inline master/internal/rm/kubernetesrm/jobs.go:2326]: question: why are we changing this to TODO?

**Review by carolinaecalderon — APPROVED** [review]
LGTM, one question about the context used in functions though

File: master/internal/rm/kubernetesrm/jobs.go (hunk 1 above)
Comment (salonig23) [reply master/internal/rm/kubernetesrm/jobs.go:2326]: in the `setupClientSetForResourceQuotaTests()` function I am mocking the list resource quota function. This function was called in getNamespaceResourceQuota with context.Background() and in setResourceQuota with context.TODO() and this inconsistency was causing the test to fail. I asked @maxrussell about it and he said ideally we would like to pass in a context, but if there isn't one context.TODO() is better in this case

**Review by carolinaecalderon — APPROVED** [review]
LGTM

**Pull request merged**